### DISCUSSION TITLE
Starting EFI over 200Mb causes everything to freeze

### DISCUSSION THREAD

**Discussion by ericspidledw** [ipxe/ipxe#1303] [Q&A]
A little stuck on this one and it appears that iPXE is doing everything properly. I have a large EFI application (~213Mb) that when loaded in stops at some point inside the `EFI StartImage` function. I know it may be a firmware bug or something else (since it is after all in the UEFI call) but I was curious if anyone had any previous experience with booting large EFI images under UEFI via iPXE. iPXE is able to allocate the RAM for the image just fine and load it just fine as well which is why I'm so confused because I would presume the image at that point should just execute as is. It is worth noting that lowering the image to about 197Mb causes it to load up just fine... so I know the image is good. Any thoughts or suggestions would be appreciated. (I see some questions in the discussions about larger images but they pertain to legacy BIOS not UEFI)

**Comment by mcb30** [comment]
Interesting.  Mostly likely the image itself is accessing memory that doesn't actually belong to it, and the large image size happens to cause the image placement in memory to end up somewhere where the illegally accessed location happens to be used for something.  Old versions of the Linux kernel had a bug like this: the kernel's `.bss` section wasn't covered by the kernel's PE headers, and so the kernel code would write to areas of memory that the UEFI firmware hadn't allocated for it.

You can test this hypothesis by deliberately doing things that will perturb the memory map: e.g. downloading additional images before and/or after your EFI application.

You can get a partial indication of what your application is doing by building iPXE with `DEBUG=efi_wrap`: this will give you `strace`-like output showing which boot services calls your application is issuing.  This may let you narrow the problem down far enough that knowledge of your application's code will let you identify the problem.  If not then, you're most likely going to have to attach a debugger to your EFI firmware.

**Reply by ericspidledw** [reply]
Thank you for the response and suggestions! I also thought it was maybe trying to use a piece of memory that it didn't have glad to hear I wasn't too far off in thinking so. I will try enabling those debug traces and loading in more images to see if I can figure out what's really going on. I'll report back once I get some more information thanks again for taking the time to respond
